matrix/matrix.py

```diff
@@ -0,0 +1,184 @@
+
+        calculated_matrix = []
+
+        for i in range(len(self.tuples)):
+
+            calculated_line = []
+            for j in range(len(self.tuples)):
+                calculated_line.append(
+                    operation(self.tuples[i][j], other.tuples[i][j]))
+
+            calculated_matrix.append(tuple(calculated_line))
+
+        return Matrix(*calculated_matrix)
+
+    def _by_scalar(self, scalar, operation):
+        """Calculate operation on matrix by scalar.
+
+        Args:
+            operation (function): the function of the operator.
+            scalar (number): the scalar to operate by.
+
+        Returns:
+            Matrix: the result of the operation.
+
+        Note:
+            Scalar must be a number. Operations with scalars are only * and /.
+        """
+        calculated_matrix = []
+
+        for i in range(len(self.tuples)):
+
+            calculated_line = []
+            for j in range(len(self.tuples)):
+                calculated_line.append(operation(self.tuples[i][j], scalar))
+
+            calculated_matrix.append(tuple(calculated_line))
+
+        return Matrix(*calculated_matrix)
+
+    def __add__(self, other):
+        """Add two matrices."""
+        return self._matrix_by_matrix(other, operator.add)
+
+    def __sub__(self, other):
+        """Subtract two matrices."""
+        return self._matrix_by_matrix(other, operator.sub)
+
+    def __mul__(self, other):
+        """Multiply two matrices or scalar by matrix."""
+        if is_number(other):
+            return self._by_scalar(other, operator.mul)
+
+        return self._matrix_by_matrix(other, operator.mul)
+
+    def __rmul__(self, other):
+        """Define to handle reverse multiplication (10 * matrix) as regular
+        multiplication."""
+        return self.__mul__(other)
+
+    def __truediv__(self, other):
+        """Divide two matrices or scalar by matrix. Returns floats if
+        needed."""
+        if not is_number(other):
+            raise TypeError(
+                f'Matrices division not supported on {type(other)}')
+
+        return self._by_scalar(other, operator.truediv)
+
+    def __floordiv__(self, other):
+        """Define to handle floor division (5 / 2 = 2) as regular
+        division."""
+        return self.__truediv__(other)
+
+    def _check_bool_matrix(self, matrix, operation):
+        """Return true if all the items in the matrix are true else false."""
+        func = {operator.eq: all, operator.ne: any}.get(operation)
+        return func(
+            [item for line_data in matrix.tuples for item in line_data])
+
+    def __eq__(self, other):
+        """Compare two matrices. Returns true if they are equal else false."""
+        return self._check_bool_matrix(
+            self._matrix_by_matrix(other, operator.eq), operator.eq)
+
+    def __ne__(self, other):
+        """Compare two matrices. Returns true if they are not equal else
+        false."""
+        return self._check_bool_matrix(
```

Comment:
Just `not self == other`
Find other places where you can reuse operators, e.g. the subtraction operator